drivers/ethernet/eth_stm32_hal_v1.c

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright (c) 2017 Erwin Rol <erwin@erwinrol.com>
+ * Copyright (c) 2020 Alexander Kozhinov <ak.alexander.kozhinov@gmail.com>
+ * Copyright (c) 2021 Carbon Robotics
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/crc.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <zephyr/net/net_pkt.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/ethernet.h>
+#include <zephyr/net/phy.h>
+#include <ethernet/eth_stats.h>
+#include <soc.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
+#include <zephyr/net/lldp.h>
+#include <zephyr/drivers/hwinfo.h>
+
+#if defined(CONFIG_NET_DSA_DEPRECATED)
+#include <zephyr/net/dsa.h>
+#endif
+
+#if defined(CONFIG_PTP_CLOCK_STM32_HAL)
+#include <zephyr/drivers/ptp_clock.h>
+#endif /* CONFIG_PTP_CLOCK_STM32_HAL */
+
+#define LOG_LEVEL CONFIG_ETHERNET_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(eth_stm32_hal_v1);
+
+#include "eth.h"
+#include "eth_stm32_hal_priv.h"
+
+static ETH_DMADescTypeDef dma_rx_desc_tab[ETH_RXBUFNB] __eth_stm32_desc;
+static ETH_DMADescTypeDef dma_tx_desc_tab[ETH_TXBUFNB] __eth_stm32_desc;
+
+static uint8_t dma_rx_buffer[ETH_RXBUFNB][ETH_STM32_RX_BUF_SIZE] __eth_stm32_buf;
+static uint8_t dma_tx_buffer[ETH_TXBUFNB][ETH_STM32_TX_BUF_SIZE] __eth_stm32_buf;
+
+int eth_tx_v1(const struct device *dev, struct net_pkt *pkt)
+{
+	struct eth_stm32_hal_dev_data *dev_data = dev->data;
+	ETH_HandleTypeDef *heth = &dev_data->heth;
+	int res;
+	size_t total_len;
+	uint8_t *dma_buffer;
+	__IO ETH_DMADescTypeDef *dma_tx_desc;
+	HAL_StatusTypeDef hal_ret = HAL_OK;
+#if defined(CONFIG_PTP_CLOCK_STM32_HAL)
+	bool timestamped_frame;
+#endif /* CONFIG_PTP_CLOCK_STM32_HAL */
+
+	__ASSERT_NO_MSG(pkt != NULL);
+	__ASSERT_NO_MSG(pkt->frags != NULL);
+
+	total_len = net_pkt_get_len(pkt);
+	if (total_len > (ETH_STM32_TX_BUF_SIZE * ETH_TXBUFNB)) {
+		LOG_ERR("PKT too big");
+		return -EIO;
+	}
+
+	k_mutex_lock(&dev_data->tx_mutex, K_FOREVER);
+
+	dma_tx_desc = heth->TxDesc;
+	while (IS_ETH_DMATXDESC_OWN(dma_tx_desc) != (uint32_t)RESET) {
+		k_yield();
+	}
+
+#if defined(CONFIG_PTP_CLOCK_STM32_HAL)
```

Comment:
CONFIG_PTP_CLOCK_STM32_HAL depends on the v2 hal, v1 doesn't support ptp

Reply:
Will clean the ptp code in v1